src/components/main-panel/main-panel.tsx

```diff
@@ -73,11 +84,30 @@ export const MainPanel = () => {
         })
     );
 
+    const handleModalOnClick = (element?: TElement) => {
+        const successToast = {
+            message: "Changes discarded.",
+            type: EToastTypes.SUCCESS_TOAST,
+        };
+        removeFocusedElement();
+        focusElement(element);
+        hideModal();
+        showToast(successToast);
```

Comment:
part of the discard changes logic is used in multiple places, can you extract the common parts to either the `useModal` hook or create a new hook?